core/scripts/chaincli/handler/debug.go

```diff
@@ -162,12 +162,18 @@ func (k *Keeper) Debug(ctx context.Context, args []string) {
 		}
 		addLink("trigger", common.ExplorerLink(chainID, txHash))
 		blockNum = receipt.BlockNumber.Uint64()
-		if len(receipt.Logs) <= int(logIndex) {
-			failCheckArgs(fmt.Sprintf("the provided transaction has %d logs but index %d was requested", len(receipt.Logs), logIndex), err)
+		// find matching log event in tx
+		var triggeringEvent *types.Log
+		for i, log := range receipt.Logs {
+			if log.Index == uint(logIndex) {
+				triggeringEvent = receipt.Logs[i]
+			}
+			if i == len(receipt.Logs) {
+				failCheckArgs(fmt.Sprintf("unable to find log with index %d in transaction", logIndex), nil)
+			}
```

Comment:
won't this condition only be true after the final iteration of this for loop? i.e., should this check instead be `if i == len(receipt.Logs) - 1`?

or another option might be to:
```
for i, log := range receipt.Logs {
    if log.Index == uint(logIndex) {
        triggeringEvent = receipt.Logs[i]
    }
}
if triggeringEvent == nil {
        failCheckArgs(fmt.Sprintf("unable to find log with index %d in transaction", logIndex), nil)
}
```

Reply:
:see_no_evil: good catch